src/Core/Settings.h

```diff
@@ -284,6 +284,7 @@ class IColumn;
     M(UInt64, read_backoff_min_concurrency, 1, "Settings to try keeping the minimal number of threads in case of slow reads.", 0) \
     \
     M(Float, memory_tracker_fault_probability, 0., "For testing of `exception safety` - throw an exception every time you allocate memory with the specified probability.", 0) \
+    M(Float, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_fault_probability, 1.0, "For testing of `PartsSplitter` - split read ranges into intersecting and non intersecting every time you read from MergeTree with the specified probability.", 0) \
```

Comment:
Shouldn't this be default to `0` and enabled for tests via `tests/config/config.d/`?

Reply:
We first need to set it to 1.0 and then fix all tests that are broken.
After this we will enable it only for tests via randomized settings.